src/orders/components/OrderRefundDialog/OrderRefundDialog.tsx

```diff
@@ -1,42 +1,79 @@
+import { useUserPermissions } from "@dashboard/auth/hooks/useUserPermissions";
 import { DashboardModal } from "@dashboard/components/Modal";
 import { RadioTiles } from "@dashboard/components/RadioTiles/RadioTiles";
+import { hasPermissions } from "@dashboard/components/RequirePermissions";
+import { OrderDetailsFragment, PermissionEnum } from "@dashboard/graphql";
 import { buttonMessages } from "@dashboard/intl";
-import { Box, Button, Text } from "@saleor/macaw-ui-next";
+import { Box, Button, Text, Tooltip } from "@saleor/macaw-ui-next";
 import React from "react";
-import { useIntl } from "react-intl";
+import { FormattedMessage, useIntl } from "react-intl";
 
 import { orderRefundDialogMesages } from "./messages";
+import { calculateOrderLineRefundTotals, isEveryLineFullyRefunded } from "./utils";
 
 interface OrderRefundDialogProps {
+  order: OrderDetailsFragment;
   open: boolean;
   onClose: () => void;
   onStandardRefund: () => void;
   onManualRefund: () => void;
 }
 
-type RefundType = "standard" | "manual";
+type RefundType = "standard" | "manual" | "none";
 
 export const OrderRefundDialog = ({
+  order,
   open,
   onClose,
   onStandardRefund,
   onManualRefund,
 }: OrderRefundDialogProps) => {
-  const [selected, setSelected] = React.useState<RefundType>("standard");
+  const [selected, setSelected] = React.useState<RefundType>("none");
   const intl = useIntl();
   const handleClose = () => {
     setSelected("standard");
     onClose();
   };
 
+  const userPermissions = useUserPermissions();
+  const canCreateManualRefund = hasPermissions(userPermissions ?? [], [
+    PermissionEnum.HANDLE_PAYMENTS,
+  ]);
+  const canCreateStandardRefund = !isEveryLineFullyRefunded(calculateOrderLineRefundTotals(order));
+  const handleChangeRefundType = (val: string) => {
+    if (val === "standard" && canCreateStandardRefund) {
+      setSelected("standard");
+    }
+
+    if (val === "manual" && canCreateManualRefund) {
+      setSelected("manual");
+    }
+  };
+
+  React.useEffect(() => {
+    if (canCreateStandardRefund) {
+      setSelected("standard");
+
+      return;
+    }
+
+    if (canCreateManualRefund) {
+      setSelected("manual");
+
+      return;
+    }
+
+    setSelected("none");
+  }, [canCreateStandardRefund, canCreateManualRefund]);
```

Comment:
suggestion: seems like god candidate to wrap into hook

Reply:
👍